src/Device.ts

```diff
@@ -0,0 +1,61 @@
+import {DeviceInterface} from "./DeviceInterface"
+import {Logger} from "./Logger";
+
+abstract class Device implements DeviceInterface {
```

Comment:
https://softwareengineering.stackexchange.com/questions/106601/in-simple-words-what-are-are-the-purposes-of-abstract-classes-and-or-interfaces